Implement planned topic: 0022-opentelemetry-plugins (#243)
temporalio/skill-temporal-developer: references/python/integrations/opentelemetry.md

> # Temporal OpenTelemetry Integration (Python)

Reviewer: This file provides very little value over what is added to observability.md. Consider just deleting it?

Reviewer: Decided to cut this file and move it just into observability.md

opentelemetry.trace.set_tracer_provider(provider)

client = await Client.connect(
    "localhost:7233",
    # Pass add_temporal_spans=True to also emit `StartWorkflow`, `RunWorkflow`,
    # `StartActivity`, `RunActivity` (and similar) spans on top of application spans.
    plugins=[OpenTelemetryPlugin()],
)

worker = Worker(
    client,
    task_queue="my-task-queue",
    workflows=[MyWorkflow],
    activities=[my_activity],
)
```

Then just use standard OpenTelemetry APIs in your workflow code — durations are accurate.

```python
from datetime import timedelta
from opentelemetry.trace import get_tracer
from temporalio import workflow

@workflow.defn
class MyWorkflow:
    @workflow.run
    async def run(self) -> None:
        tracer = get_tracer(__name__)
        with tracer.start_as_current_span("workflow-operation"):
            await workflow.execute_activity(
                my_activity,
                start_to_close_timeout=timedelta(seconds=30),
            )
```

## Common mistakes

- **Registering the same plugin on both Client and Worker.** Register on the Client only; Workers inherit.

> - **Calling `Client.connect` before `opentelemetry.trace.set_tracer_provider(provider)`.** With `OpenTelemetryPlugin`, the workflow interceptor factory raises `ValueError("When using OpenTelemetryPlugin, the global trace provider must be a ReplaySafeTracerProvider. Use init_tracer_provider to create one.")`.

Reviewer 2: "init_tracer_provider" is actually a bug in the error message. It got renamed.

Reviewer 2: https://github.com/temporalio/sdk-python/pull/1584

> - **Calling `Client.connect` before `opentelemetry.trace.set_tracer_provider(provider)`.** With `OpenTelemetryPlugin`, the workflow interceptor factory raises `ValueError("When using OpenTelemetryPlugin, the global trace provider must be a ReplaySafeTracerProvider. Use init_tracer_provider to create one.")`.

Reviewer: ```suggestion
- **Calling `Client.connect` before `opentelemetry.trace.set_tracer_provider(provider)`.**`OpenTelemetryPlugin`, will raise an exception if you do this
```

- **Building a plain `opentelemetry.sdk.trace.TracerProvider` and passing it to `set_tracer_provider`.** `OpenTelemetryPlugin` requires `ReplaySafeTracerProvider` — build it via `create_tracer_provider(...)`.

> - **Adding `with_passthrough_modules("opentelemetry")` to a `SandboxedWorkflowRunner` manually.** The plugin already does this.

Reviewer: ```suggestion
```

> - **Adding `with_passthrough_modules("opentelemetry")` to a `SandboxedWorkflowRunner` manually.** The plugin already does this.

Reviewer 2: Doesn't hurt though, so not sure it qualifies as a common mistake.
